app/src/main/java/ch/epfl/cs311/wanderwave/model/spotify/SpotifyController.kt

```diff
@@ -305,6 +307,87 @@ constructor(
     }
   }
 
+  /**
+   * Retrieve the image URL of an album.
+   *
+   * @param albumId The Spotify ID of the album.
+   * @return A Flow that emits the album image URL as a string.
+   */
+  fun getAlbumImage(albumId: String): Flow<String?> {
+    return callbackFlow {
+      val accessToken = authenticationController.getAccessToken()
+
+      if (accessToken == null) {
+        trySend(null)
+        close()
+        return@callbackFlow
+      }
```

Comment:
You don't need to get the `accessToken` here, `spotifyGetFromURL` (l.393)  already handles it

Reply:
Done